app/components/form/SideModalForm.tsx

```diff
@@ -14,8 +14,38 @@ import type { ApiError } from '@oxide/api'
 import { Button } from '~/ui/lib/Button'
 import { SideModal } from '~/ui/lib/SideModal'
 
+type ResourceName =
+  | 'disk'
+  | 'floating IP'
+  | 'identity provider'
+  | 'image'
+  | 'IP pool'
+  | 'IP range'
+  | 'network interface'
+  | 'project'
+  | 'project image'
+  | 'role'
+  | 'rule'
+  | 'silo'
+  | 'silo image'
+  | 'snapshot'
+  | 'SSH key'
+  | 'subnet'
+  | 'VPC'
```

Comment:
I think this should just be a string. I'm not worried about typoing `projec` or whatever.